bin/propolis-server/src/lib/server.rs

```diff
@@ -318,36 +320,111 @@ fn instance_spec_from_request(
     Ok(VersionedInstanceSpec::V0(spec_builder.finish()))
 }
 
-/// Attempts to register an Oximeter server reporting metrics from a new
-/// instance, returning the producer registry from that server on success.
-async fn register_oximeter(
-    server_context: &DropshotEndpointContext,
-    cfg: &MetricsEndpointConfig,
-    vm_id: uuid::Uuid,
+/// Register an Oximeter server reporting metrics from a new instance.
+///
+/// This spawns a tokio task which will indefinitely attempt to register with
+/// Nexus as an Oximeter metric producer. Once the registration succeeds, it
+/// will also arrange for the statistics produced for the managed instance to be
+/// collected.
```

Comment:
Having this loop forever, with no means of the rest of propolis-server to signal its desired termination seems a bit off to me.  Perhaps this task can be bundled into a consolidated oximeter state in `ServiceProviders`?

Reply:
Done in d61fd69